src/pkg/reg/adapter/dockerhub/adapter.go

```diff
@@ -124,6 +126,55 @@ func getAdapterInfo() *model.AdapterPattern {
 	return info
 }
 
+// Rate-limit aware wrapper function for client.Do()
+// - Avoids being hit by limit by pausing requests when less than 'lowMark' requests remaining.
+// - Pauses for given time when limit is hit.
+// - Allows 2 more attempts before giving up.
+// Reason: Observed (02/2024) penalty for hitting the limit is 120s, normal reset is 60s,
+// so it is better to not hit the wall.
+func (a *adapter) limitAwareDo(method string, path string, body io.Reader) (*http.Response, error) {
```

Comment:
- missing test case
- to much logging into **info**, I would put all in debug. I we can handle the rate limit there is nothing to info about.
- I would maybe allow one info statement, when we reach/hit the limit. Other informatio is relevant for debut only.